src/legend/LegendItem.store.js

```diff
@@ -72,10 +72,10 @@ const formFieldsConfigs = [{
     }), ColorPicker),
 }];
 
-
 // Called when a field is changed
 export function onFieldChange(fieldName, value) {
-    const model = legendItemStore.getState().model;
+    // Shallow clone model, so the assignment below doesn't mutate the original
+    const model = { ...legendItemStore.getState().model };
```

Comment:
If this is in fact a d2 Model instance, this could be problematic. This shallow clone will be a POJO containing only the instance's properties, but not the class methods.

In this past I have encountered a similar problem and solved it differently. I believe like this:
```javascript
const originalModel = legendItemStore.getState().model;
// This creates a fresh model instance with all the properties of the old model 
// and all of the class methods are still there
const model = originalModel.modelDefinition.create(originalModel..toJSON());
model[fieldName] = value
```

Reply:
> If this is in fact a d2 Model instance

I believe it is actually, hmm.

Reply:
This code is so full of footguns. I'll check the prototype of the model to verify, thanks for the heads up.

Reply:
I've checked, but it doesn't seem to be an instance of a d2 Model:

```
Object.getPrototypeOf(originalModel)
> Object { … }
```

So that's good 🙂

Reply:
That's good 👍